integration-tests/tests/client/process_blocks.rs

```diff
@@ -1558,6 +1575,34 @@ fn test_gc_after_state_sync() {
     assert!(env.clients[1].chain.clear_data(tries, 2).is_ok());
 }
 
+#[test]
+fn test_num_blocks_in_storage_config_setting() {
+    let epoch_length = 128;
+    // TODO - figured out how to change `num_epochs_to_keep_store_data = 10`
+    let mut genesis = Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+    genesis.config.epoch_length = epoch_length;
+    let mut chain_genesis = ChainGenesis::test();
+    chain_genesis.epoch_length = epoch_length;
+    let num_epochs_to_keep_store_data = 10;
+    let mut env = TestEnv::builder(chain_genesis)
+        .clients_count(2)
+        .runtime_adapters(create_nightshade_runtimes_with_num_epochs(
+            &genesis,
+            2,
+            Some(num_epochs_to_keep_store_data),
+        ))
+        .build();
+    for i in 1..epoch_length * (num_epochs_to_keep_store_data - 1) + 2 {
+        let block = env.clients[0].produce_block(i).unwrap().unwrap();
+        env.process_block(0, block.clone(), Provenance::PRODUCED);
+        env.process_block(1, block, Provenance::NONE);
+    }
+    // Should crash, but doesn't?
+    for i in 4..epoch_length * (num_epochs_to_keep_store_data - 1) + 2 {
+        env.clients[0].chain.get_block_by_height(i).unwrap();
```

Comment:
Does this test pass? 

Should you check the first few blocks are gc'ed?

Reply:
@mzhangmzz This passes the test, but
Works:
``` (num_epochs_to_keep_store_data - 1) + 2```
Doesn't pass:
``` (num_epochs_to_keep_store_data) + 2```

You are right, I should also add a negative check in the test case.

Reply:
@mzhangmzz I updated the test, to check whenever blocks get GC'ed